roles/jd-server/src/main.rs

```diff
@@ -108,163 +103,5 @@ async fn main() {
-
-    let cloned = config.clone();
-    let mempool_cloned = mempool.clone();
-    let (sender_add_txs_to_mempool, receiver_add_txs_to_mempool) = unbounded();
-    task::spawn(async move {
-        JobDeclarator::start(
-            cloned,
-            sender,
-            mempool_cloned,
-            new_block_sender,
-            sender_add_txs_to_mempool,
-        )
-        .await
-    });
-    task::spawn(async move {
-        loop {
-            if let Ok(add_transactions_to_mempool) = receiver_add_txs_to_mempool.recv().await {
-                let mempool_cloned = mempool.clone();
-                task::spawn(async move {
-                    match lib::mempool::JDsMempool::add_tx_data_to_mempool(
-                        mempool_cloned,
-                        add_transactions_to_mempool,
-                    )
-                    .await
-                    {
-                        Ok(_) => (),
-                        Err(err) => {
-                            // TODO
-                            // here there should be a better error management
-                            mempool::error::handle_error(&err);
-                        }
-                    }
-                });
-            }
-        }
-    });
-
-    // Start the error handling loop
-    // See `./status.rs` and `utils/error_handling` for information on how this operates
-    loop {
-        let task_status = select! {
-            task_status = status_rx.recv() => task_status,
-            interrupt_signal = tokio::signal::ctrl_c() => {
-                match interrupt_signal {
-                    Ok(()) => {
-                        info!("Interrupt received");
-                    },
-                    Err(err) => {
-                        error!("Unable to listen for interrupt signal: {}", err);
-                        // we also shut down in case of error
-                    },
-                }
-                break;
-            }
-        };
-        let task_status: status::Status = task_status.unwrap();
-
-        match task_status.state {
-            // Should only be sent by the downstream listener
-            status::State::DownstreamShutdown(err) => {
-                error!(
-                    "SHUTDOWN from Downstream: {}\nTry to restart the downstream listener",
-                    err
-                );
-            }
-            status::State::TemplateProviderShutdown(err) => {
-                error!("SHUTDOWN from Upstream: {}\nTry to reconnecting or connecting to a new upstream", err);
-                break;
-            }
-            status::State::Healthy(msg) => {
-                info!("HEALTHY message: {}", msg);
-            }
-            status::State::DownstreamInstanceDropped(downstream_id) => {
-                warn!("Dropping downstream instance {} from jds", downstream_id);
-            }
-        }
-    }
+    lib::core::JDServer::new(config).start().await;
```

Comment:
Need associated import.
```suggestion
    JDServer::new(config).start().await;
```